File: frontend/src/api/campaign.ts
import api from "../services/api";

export interface Campaign {
  id?: number;
  title: string;
  description?: string;
  budget: number;
  start_date: string;
  end_date: string;
  status?: "DRAFT" | "ACTIVE" | "COMPLETED" | "CANCELLED";
  company_id: number;
  created_at?: string;
  updated_at?: string;
}

export const getCampaigns = async () => {
  const res = await api.get("/campaigns");
  return res.data;
};

export const getCampaign = async (id: number) => {
  const res = await api.get(`/campaigns/${id}`);
  return res.data;
};

export const createCampaign = async (data: Campaign) => {
  const res = await api.post("/campaigns", data);
  return res.data;
};

export const updateCampaign = async (id: number, data: Partial<Campaign>) => {
  const res = await api.put(`/campaigns/${id}`, data);
  return res.data;
};

export const deleteCampaign = async (id: number) => {
  const res = await api.delete(`/campaigns/${id}`);
  return res.data;
};

File: backend/app/schemas/campaign.py
from datetime import date, datetime
from decimal import Decimal
from enum import Enum

from pydantic import BaseModel, ConfigDict


class CampaignStatus(str, Enum):
    DRAFT = "DRAFT"
    ACTIVE = "ACTIVE"
    COMPLETED = "COMPLETED"
    CANCELLED = "CANCELLED"


class CampaignBase(BaseModel):
    title: str
    description: str | None = None
    budget: Decimal
    start_date: date
    end_date: date
    status: CampaignStatus = CampaignStatus.DRAFT
    company_id: int


class CampaignCreate(CampaignBase):
    pass


class CampaignUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    budget: Decimal | None = None
    start_date: date | None = None
    end_date: date | None = None
    status: CampaignStatus | None = None
    company_id: int | None = None


class CampaignResponse(CampaignBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)